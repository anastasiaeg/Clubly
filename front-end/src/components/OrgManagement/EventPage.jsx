import React from "react";
import { Link } from "react-router-dom";
import {
  AnchorButton,
  Button,
  Classes,
  Dialog,
  H2,
  H5,
  Intent,
  InputGroup
} from "@blueprintjs/core";
import { DateRangePicker, TimePrecision } from "@blueprintjs/datetime";
import "./orgmanagement.css";
import * as routes from "../../constants/routes";
import { eventsService } from "../../services/eventsService";
import Loading from "../Common/Loading";

export default class EventPage extends React.Component {
  emptyEvent = {
    name: "",
    description: "",
    location: "",
    startTime: undefined,
    endTime: undefined
  };
  constructor(props) {
    super(props);
    this.state = {
      editDialogOpened: false,
      eventForm: this.emptyEvent,
      loading: true,
      editing: false,
      eventError: "",
      formError: "",
      event: {}
    };
  }

  componentDidMount() {
    this.loadData();
  }

  loadData = async () => {
    const { event_id } = this.props.route.match.params;
    eventsService.getEvent(event_id).then(result => {
      if (result.data) {
        this.event = result.data;
        this.setState({
          loading: false,
          event: result.data,
          eventError: ""
        });
      } else {
        this.setState({
          loading: false,
          eventError: result.error
        });
      }
    });
  };

  onChangeDateTime = selectedDates => {
    if (!selectedDates[0] || !selectedDates[1]) {
      this.setState({
        formError: "Date Form can't be null"
      });
    } else {
      let eventForm = Object.assign({}, this.state.eventForm);
      eventForm.startTime = selectedDates[0];
      eventForm.endTime = selectedDates[1];
      this.setState({
        eventForm
      });
    }
  };

  toggleEventDialog = event => {
    this.setState({
      editDialogOpened: true,
      eventForm: event !== null ? event : this.state.eventForm,
      editing: event !== null,
      formError: ""
    });
  };

  handleCloseDialog = () => {
    this.setState({
      editDialogOpened: false,
      editing: false,
      eventForm: this.emptyEvent,
      formError: ""
    });
  };

  submit = () => {
    console.log("submit");
  };

  formChange = (attribute, value) => {
    let eventForm = Object.assign({}, this.state.eventForm);
    eventForm[attribute] = value;
    this.setState({ eventForm });
  };

  render() {
    const { width } = this.props;
    const {
      editDialogOpened,
      loading,
      event,
      editing,
      eventForm,
      formError,
      eventError
    } = this.state;
    if (loading) {
      return <Loading />;
    } else if (eventError) {
      return <div>Error with loading Event</div>;
    }
    return (
      <div>
        <div className="row">
          {/* {event.image && ( */}
          <div className="org-image-container">
            <img
              className="org-image"
              src={
                event.image ||
                "https://www.belfercenter.org/themes/belfer/images/event-default-img-med.png"
              }
              alt={event.name + " Logo"}
            />
          </div>
          {/* )} */}
          <div>
            <H2>
              {event.name}
              <Button icon="cog" onClick={() => this.toggleEventDialog(null)} />
            </H2>
            <div
              className="org-description"
              style={{ width: width - 538 + "px" }}
            >
              {event.description}
            </div>
            <br />
            <div className="org-tags">
              Tags:
              {event.tags.map(tag => (
                <span key={"tag" + tag.id}>
                  <Link to={routes.SEARCH}>
                    <span className="bp3-tag .modifier">{tag.name}</span>
                  </Link>
                </span>
              ))}
            </div>
            {new Date(event.startTime).getTime() < new Date().getTime() && (
              <AnchorButton text={"RSVP"} />
            )}
          </div>
        </div>
        <br />
        <Dialog
          autoFocus={true}
          isOpen={editDialogOpened}
          onClose={this.handleCloseDialog}
          title={editing ? "Edit the Event" : "Create New Events"}
          canEscapeKeyClose={true}
        >
          <div className={Classes.DIALOG_BODY}>
            <H5>Name</H5>
            <InputGroup
              large={true}
              round={true}
              fill={true}
              value={eventForm.name}
              onChange={event => this.formChange("name", event.target.value)}
              placeholder="Enter name..."
            />
            <br />
            <H5>Description</H5>
            <textarea
              className="bp3-input bp3-round .modifier"
              dir="auto"
              rows="10"
              cols="1000"
              value={eventForm.description}
              onChange={event =>
                this.formChange("description", event.target.value)
              }
              placeholder="Enter description..."
            />
            <br />
            <br />
            <H5>Location</H5>
            <InputGroup
              large={true}
              round={true}
              fill={true}
              required={true}
              value={eventForm.location}
              onChange={event =>
                this.formChange("location", event.target.value)
              }
              placeholder="Enter location..."
            />
            <br />
            <H5>Start Time - End Time</H5>
            <DateRangePicker
              shortcuts={false}
              className={Classes.ELEVATION_1}
              allowSingleDayRange={true}
              timePrecision={TimePrecision.MINUTE}
              onChange={this.onChangeDateTime}
              timePickerProps={{
                useAmPm: true
              }}
              value={[eventForm.startTime, eventForm.endTime]}
            />
          </div>
          <div className={Classes.DIALOG_FOOTER + " row"}>
            {formError}
            <div>
              {editing && (
                <AnchorButton
                  intent={Intent.DANGER}
                  onClick={this.handleCloseDialog}
                >
                  Remove Event
                </AnchorButton>
              )}
            </div>
            <div className="push">
              <Button onClick={this.handleCloseDialog}>Cancel</Button>
            </div>
            <div>
              <AnchorButton intent={Intent.SUCCESS} onClick={this.submit}>
                Save
              </AnchorButton>
            </div>
          </div>
        </Dialog>
      </div>
    );
  }
}
