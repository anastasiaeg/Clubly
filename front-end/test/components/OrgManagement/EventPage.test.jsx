import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import EventPage from '../../../src/components/OrgManagement/EventPage';

const props = {
    event: {
        name: "Resume panel",
        description: "Come get tips and ask questions about resumes and cover letters. Representatives from Advance Auto Parts and RepublicWireless will be there to answer questions and look over resumes, as well as Leslie Rand-Pickett from the CSC graduate career services",
        startTime: new Date(2018, 11, 24, 10, 33, 30, 0),
        endTime: new Date(2018, 11, 24, 10, 33, 30, 0),
        location: "Daniels Hall",
        tags: [
            "Technology",
            "Panel"
        ],
        rsvp: [
            "Anastasia",
            "This person",
            "next person"
        ],
        attended: []
    }
}

describe('Event Page', () => {
    test('should render', () => {
        const component = renderer.create(
            <MemoryRouter>
                <EventPage event={props.event}/>
            </MemoryRouter>
        )
        .toJSON();
        expect(component).toMatchSnapshot();
    });
})