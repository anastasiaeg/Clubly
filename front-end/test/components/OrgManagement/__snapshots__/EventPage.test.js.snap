// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Page should render 1`] = `
<div
  className="event-short"
>
  <div
    className="bp3-card bp3-elevation-0"
  >
    <a
      href="/event"
      onClick={[Function]}
    >
      <h4
        className="bp3-heading"
      >
        Resume panel
      </h4>
    </a>
    <p
      className="bp3-text-overflow-ellipsis"
      title={undefined}
    >
      Come get tips and ask questions about resumes and cover letters. Representatives from Advance Auto Parts and RepublicWireless will be there to answer questions and look over resumes, as well as Leslie Rand-Pickett from the CSC graduate career services
    </p>
    <div>
      Monday, December 24, 10:33 AM
    </div>
    <div>
      <a
        href="https://www.google.com/maps/search/?api=1&query=Daniels%20Hall"
        target="_blank"
      >
        <span
          className="bp3-icon bp3-icon-locate"
          icon="locate"
        >
          <svg
            data-icon="locate"
            fill={undefined}
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              locate
            </desc>
            <path
              d="M15 7h-.09A6.98 6.98 0 0 0 9 1.1V1c0-.55-.45-1-1-1S7 .45 7 1v.09A6.98 6.98 0 0 0 1.1 7H1c-.55 0-1 .45-1 1s.45 1 1 1h.1A6.969 6.969 0 0 0 7 14.91V15c0 .55.45 1 1 1s1-.45 1-1v-.09A6.98 6.98 0 0 0 14.9 9h.1c.55 0 1-.45 1-1s-.45-1-1-1zm-6.02 5.9c-.05-.5-.46-.9-.98-.9s-.93.4-.98.9A5.017 5.017 0 0 1 3.1 8.98c.5-.05.9-.46.9-.98s-.4-.93-.9-.98A5.017 5.017 0 0 1 7.02 3.1c.05.5.46.9.98.9s.93-.4.98-.9c1.97.39 3.52 1.95 3.92 3.92-.5.05-.9.46-.9.98s.4.93.9.98a5.017 5.017 0 0 1-3.92 3.92zM8 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </a>
      Daniels Hall
    </div>
    <br />
    <span
      className="org-tags"
    >
      <a
        href="/search"
        onClick={[Function]}
      >
        <span
          className="bp3-tag .modifier"
        >
          Technology
        </span>
      </a>
    </span>
    <span
      className="org-tags"
    >
      <a
        href="/search"
        onClick={[Function]}
      >
        <span
          className="bp3-tag .modifier"
        >
          Panel
        </span>
      </a>
    </span>
  </div>
  <br />
</div>
`;
